cmd/polisatomicswap/main.go

```diff
@@ -0,0 +1,1142 @@
+	}
+	params := []json.RawMessage{param0, param1}
+	rawResp, err := c.RawRequest("fundrawtransaction", params)
+	if err != nil {
+		return nil, 0, err
+	}
+	var resp struct {
+		Hex       string  `json:"hex"`
+		Fee       float64 `json:"fee"`
+		ChangePos float64 `json:"changepos"`
+	}
+	err = json.Unmarshal(rawResp, &resp)
+	if err != nil {
+		return nil, 0, err
+	}
+	fundedTxBytes, err := hex.DecodeString(resp.Hex)
+	if err != nil {
+		return nil, 0, err
+	}
+	fundedTx = &wire.MsgTx{}
+	err = fundedTx.Deserialize(bytes.NewReader(fundedTxBytes))
+	if err != nil {
+		return nil, 0, err
+	}
+	feeAmount, err := gopolisutil.NewAmount(resp.Fee)
+	if err != nil {
+		return nil, 0, err
+	}
+	return fundedTx, feeAmount, nil
+}
+
+// getFeePerKb queries the wallet for the transaction relay fee/kB to use and
+// the minimum mempool relay fee.  It first tries to get the user-set fee in the
+// wallet.  If unset, it attempts to find an estimate using estimatefee 6.  If
+// both of these fail, it falls back to mempool relay fee policy.
+func getFeePerKb(c *rpc.Client) (useFee, relayFee gopolisutil.Amount, err error) {
+	var netInfoResp struct {
+		RelayFee float64 `json:"relayfee"`
+	}
+	var walletInfoResp struct {
+		PayTxFee float64 `json:"paytxfee"`
+	}
+	var estimateResp struct {
+		FeeRate float64 `json:"feerate"`
+	}
+
+	netInfoRawResp, err := c.RawRequest("getnetworkinfo", nil)
+	if err == nil {
+		err = json.Unmarshal(netInfoRawResp, &netInfoResp)
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+	walletInfoRawResp, err := c.RawRequest("walletinfo", nil)
+	if err == nil {
+		err = json.Unmarshal(walletInfoRawResp, &walletInfoResp)
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+
+	relayFee, err = gopolisutil.NewAmount(netInfoResp.RelayFee)
+	if err != nil {
+		return 0, 0, err
+	}
+	payTxFee, err := gopolisutil.NewAmount(walletInfoResp.PayTxFee)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	// Use user-set wallet fee when set and not lower than the network relay
+	// fee.
+	if payTxFee != 0 {
+		maxFee := payTxFee
+		if relayFee > maxFee {
+			maxFee = relayFee
+		}
+		return maxFee, relayFee, nil
+	}
+
+	params := []json.RawMessage{[]byte("6")}
+	estimateRawResp, err := c.RawRequest("estimatesmartfee", params)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	err = json.Unmarshal(estimateRawResp, &estimateResp)
+	if err == nil && estimateResp.FeeRate > 0 {
+		useFee, err = gopolisutil.NewAmount(estimateResp.FeeRate)
+		if relayFee > useFee {
+			useFee = relayFee
+		}
+		return useFee, relayFee, err
+	}
+
+	fmt.Println("warning: falling back to mempool relay fee policy")
+	return relayFee, relayFee, nil
+}
+
+// getRawChangeAddress calls the getrawchangeaddress JSON-RPC method.  It is
+// implemented manually as the rpcclient implementation always passes the
+// account parameter which was removed in Polis Core 0.15.
+func getRawChangeAddress(c *rpc.Client) (gopolisutil.Address, error) {
+	params := []json.RawMessage{[]byte(`"legacy"`)}
```

Comment:
Your getrawchangeaddress method does not take any optional parameters, and your wallet does not have segwit support (which is what legacy means here for bitcoin core -- it requests a P2PKH address instead of P2SH-witness or P2WPKH).

Reply:
Removed the "legacy" param, plain getrawchangeaddress seems to be working because we don't have segwit support.